### PR TITLE
DOC: Add Timedelta accepting float value

### LINKED ISSUE #60044: DOC: The `value` parameter of `pandas.Timedelta` can also accept float
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/dev/reference/api/pandas.Timedelta.html#pandas.Timedelta

### Documentation problem

It currently states:

https://github.com/pandas-dev/pandas/blob/2a10e04a099d5f1633abcdfbb2dd9fdf09142f8d/pandas/_libs/tslibs/timedeltas.pyx#L1867

There is no `float`.

### Suggested fix for documentation

Add `float, ` to accepted types. 

There already exists a code branch for `float` values so this should not be a problem:
https://github.com/pandas-dev/pandas/blob/2a10e04a099d5f1633abcdfbb2dd9fdf09142f8d/pandas/_libs/tslibs/timedeltas.pyx#L2056

### PR DESCRIPTION
**Pull request by ZKaoChi**
- [x] closes #60044

### CODE REVIEW

**Comment by 5j9** [comment]
@ZKaoChi , See [the error message for the failed test](https://results.[pre-commit.ci](https://results.pre-commit.ci/run/github/858127/1729081421.P09tMa7sRx2mXWMiXpg8cw)/run/github/858127/1729081421.P09tMa7sRx2mXWMiXpg8cw):

```
cython-lint.............................................................................................Failed
- hook id: cython-lint
- exit code: 1

/code/pandas/_libs/tslibs/timedeltas.pyx:1871:89: E501 line too long (149 > 88 characters)
```

(Keep lines below 89 characters.)

**Comment by ZKaoChi** [comment]
@5j9 I see. Thanks a lot!!!

**Review by 5j9 — COMMENTED** [review]
Another suggestion: I'd use a more descriptive commit title. Something that gives more information about the issue at hand rather than just referring to issue number.

File: pandas/_libs/tslibs/timedeltas.pyx [hunk 1]
```diff
@@ -1864,10 +1864,12 @@ class Timedelta(_Timedelta):
...
     Parameters
     ----------
-    value : Timedelta, timedelta, np.timedelta64, str, or int
+    value : Timedelta, timedelta, np.timedelta64, str, int or float
         Input value.
     unit : str, default 'ns'
-        Denote the unit of the input, if input is an integer.
+        If input is an integer, denote the unit of the input.
+        If input is a float, denote the unit of the integer parts.
+        (The decimal parts with an accuracy higher than 1 nanosecond are ignored.)
```
Comment (rhshadrach) [inline pandas/_libs/tslibs/timedeltas.pyx:1872]: I think "accuracy" would be better termed "resolution", and in which case, it's decimal parts with a resolution lower than 1 nanosecond are ignored. I also think the parentheses can be removed.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Looking good! small request.

File: pandas/_libs/tslibs/timedeltas.pyx (hunk 1 above)
Comment (ZKaoChi) [reply pandas/_libs/tslibs/timedeltas.pyx:1872]: Thanks a lot for your comment. I'm sorry I reinstalled the system so I lost the pull request. I opened a new pull request to change it. Thanks again !!!

Comment (rhshadrach) [reply pandas/_libs/tslibs/timedeltas.pyx:1872]: This is fine for now, but in the future you did not lose the branch nor pull request. It is still available:

https://github.com/ZKaoChi/pandas/branches

You can get it by `git switch origin Solution-for-issue-#60044`

Comment (ZKaoChi) [reply pandas/_libs/tslibs/timedeltas.pyx]: I've changed the wording under this pull request. Could you please review it again?

**Review by rhshadrach — APPROVED** [review]
lgtm

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @ZKaoChi

**Comment by ZKaoChi** [comment]
Thanks a lot for your help.

**Pull request merged**